src/Disks/IO/CachedOnDiskReadBufferFromFile.cpp

```diff
@@ -616,6 +616,10 @@ void CachedOnDiskReadBufferFromFile::predownload(FileSegmentPtr & file_segment)
                     continue_predownload = false;
                 }
             }
+            else
+            {
+                file_segment->completeWithState(FileSegment::State::PARTIALLY_DOWNLOADED_NO_CONTINUATION);
```

Comment:
Why do we set it only when `reserve` failed, but not `writeCache`?

Reply:
See https://github.com/ClickHouse/ClickHouse/pull/48636#discussion_r1162680871.